eos-tech-support/eos-write-image

```diff
@@ -81,53 +82,59 @@ DD_IN_OPTS+=(bs="$BLOCK_SIZE")
 DD_OUT_OPTS+=(bs="$BLOCK_SIZE")
 
 if [ "$USERID" != "0" ]; then
-    echo "Program requires superuser privileges"
+    echo "Program requires superuser privileges" >&2
     exit 1
 fi
 
 if [ ! -f "$IMAGE" ]; then
-    echo "$IMAGE not found"
+    echo "$IMAGE not found" >&2
     exit 1
 fi
 
-if [ ! -b $DEVICE ]; then
-    echo "$DEVICE not found or is not a block device"
-    exit 1
-fi
+cat_image() {
+    if file --mime-type "$IMAGE" | grep -q gzip$; then
+        # Image is gzipped
+        # The following would calculate the original size
+        # of the unzipped file if less than 4G:
+        # IMAGE_SIZE=$(zcat -l "$IMAGE" | awk 'NR==2 { print $2 }')
+        # But this doesn't help for large images,
+        # so we just show relative progress unless .size is present
+        if [ -f "$IMAGE.size" ]; then
+            IMAGE_SIZE=$(cat $IMAGE.size)
+            PV_ARGS="-s $IMAGE_SIZE"
+        fi
+        zcat "$IMAGE" | pv $PV_ARGS
+    else
+        # Image is not gzipped
+        # We can show progress as percentage of total image size
+        IMAGE_SIZE=$(ls -l "$IMAGE" | awk '{ print $5 }')
+        dd if="$IMAGE" "${DD_IN_OPTS[@]}" | pv -s ${IMAGE_SIZE}
+    fi
+}
 
-if grep -qs $DEVICE /proc/mounts; then
-    # Protect against overwriting the device currently in use
-    echo "$DEVICE is currently in use -- please unmount and try again"
-    exit 1
-fi
+if [ $DEVICE == "-" ]; then
+    cat_image
+else
+    if [ ! -b $DEVICE ]; then
+        echo "$DEVICE not found or is not a block device" >&2
+        exit 1
+    fi
 
-if ! $FORCE; then
-    read -p "Are you sure you want to overwrite all data on $DEVICE? [y/N] "
-    response=${REPLY,,} # to lower
-    if [[ ! $response =~ ^(yes|y)$ ]]; then
+    if grep -qs $DEVICE /proc/mounts; then
+        # Protect against overwriting the device currently in use
+        echo "$DEVICE is currently in use -- please unmount and try again" >&2
         exit 1
     fi
-fi
 
-# Try to format / discard all the sectors on the device
-blkdiscard "$DEVICE" &>/dev/null || true
-
-if file --mime-type "$IMAGE" | grep -q gzip$; then
-    # Image is gzipped
-    # The following would calculate the original size
-    # of the unzipped file if less than 4G:
-    # IMAGE_SIZE=$(zcat -l "$IMAGE" | awk 'NR==2 { print $2 }')
-    # But this doesn't help for large images,
-    # so we just show relative progress unless .size is present
-    if [ -f "$IMAGE.size" ]; then
-        IMAGE_SIZE=$(cat $IMAGE.size)
-        PV_ARGS="-s $IMAGE_SIZE"
+    if ! $FORCE; then
```

Comment:
Since `$FORCE` is controlled within this script (it's either `true` or `false`), then I think this usage is entirely appropriate and more direct than testing the string.